ui/de-lib/src/main/java/org/iplantc/de/admin/desktop/client/permIdRequest/presenter/PermanentIdRequestPresenter.java

```diff
@@ -131,6 +136,41 @@ public void onSuccess(String result) {
     }
 
     @Override
+    public void onUpdateRequest() {
+        getRequestDetails(new AsyncCallback<String>() {
+            @Override
+            public void onFailure(Throwable caught) {
+                view.unmask();
+                IplantErrorDialog ied = new IplantErrorDialog(I18N.DISPLAY.error(), caught.getMessage());
+                ied.show();
+            }
+
+            @Override
+            public void onSuccess(String result) {
+                view.unmask();
+                final AutoBean<PermanentIdRequesDetails> decode =
+                        AutoBeanCodex.decode(factory, PermanentIdRequesDetails.class, result);
```

Comment:
This AutoBean decoding could be done in the service facade with an `AsyncCallbackConverter`.